### ISSUE TITLE
VPN Gateway config with `activeActive: true` fails because only one public IP address gets created

### ISSUE THREAD

**Issue by SvenAelterman** [Azure/ALZ-Bicep#768]
### What happened? Provide a clear and concise description of the bug, including deployment details.

Deploying hub networking module with the following config for the VPN gateway:

```bicep
param parVpnGatewayConfig = {
  name: 'vpn-prd-vgw-scus-tasb-01'
  gatewayType: 'Vpn'
  sku: 'VpnGw1AZ'
  vpnType: 'RouteBased'
  generation: 'Generation1'
  enableBgp: false
  activeActive: true
   enableBgpRouteTranslationForNat: false
   enableDnsForwarding: false
   vpnClientConfiguration: {}
}
```

The deployed failed with
```
Gateway has 1 IP configurations specified for gateway <snip>. ActiveActive Gateway must be created with two IP configurations. (Code: ActiveActiveGatewayMustHaveExactlyTwoIpConfigurations)
```

### Please provide the correlation id associated with your error or bug.

5abecd9e-1c80-4639-ae5e-4f1d7958801d

### What was the expected outcome?

Two public IPs for the VPN gateway.

### Relevant log output

```Shell
Gateway has 1 IP configurations specified for gateway <snip>. ActiveActive Gateway must be created with two IP configurations. (Code: ActiveActiveGatewayMustHaveExactlyTwoIpConfigurations)
```

### Check previous GitHub issues

- [X] I have searched the issues for this item and found no duplicate

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by oZakari** [comment]
Hi @SvenAelterman, you are correct! I am surprised as you do appear to be the first person to call this out, and it looks like it has been an issue from early on looking at the commit history.

I have added it to the backlog. Thanks for calling it out 👍🏼

**Comment by SvenAelterman** [comment]
Quick examination of the code seems to indicate that there is a loop for creating public IP addresses, but it enumerates the gateways to be deployed, which is odd, because an ER gateway doesn't need a public IP. Hopefully the fix is simple.

**Comment by jtracey93** [comment]
Clarification: ER Gateways do need a public IP but its never used by the client, its used for the ExR PG team to manage the gateways as a PaaS Service

**Comment by SvenAelterman** [comment]
Ah, my mistake. Thanks for the clarification.

**Comment by oZakari** [comment]
Closed with #863
